File: Python/0251-0300/278. First Bad Version.py
# https://leetcode.com/problems/first-bad-version/

class Solution:
    def firstBadVersion(self, n: int) -> int:
        first, last = 1, n

        while(first <= last):
            mid = first + (last - first) // 2

            if isBadVersion(mid):
                last = mid - 1
            else:
                first = mid + 1

        return first

File: Java/0251-0300/278. First Bad Version.java
// https://leetcode.com/problems/first-bad-version/

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int first = 1, last = n;

        while(first <= last){
            int mid = first + (last - first) / 2;

            if(isBadVersion(mid)){
                last = mid - 1;
            }else{
                first = mid + 1;
            }
        }

        return first;
    }
}
